src/providers/postgres/qgspostgresprovider.cpp

```diff
@@ -1126,36 +1126,58 @@ bool QgsPostgresProvider::loadFields()
+  //                                         type TEXT NOT NULL, config TEXT,
+  //                                         PRIMARY KEY(schema_name, table_name, field_name));
+  const QString sql = QStringLiteral( "SELECT field_name, type, config "
+                                      "FROM %1 WHERE schema_name = %2 "
+                                      "AND table_name = %3 "
+                                      "AND field_name IN ( %4 )" ) .
+                      arg( EDITOR_WIDGET_STYLES_TABLE, quotedValue( mSchemaName ),
+                           quotedValue( mTableName ), quotedFnames.join( "," ) );
+  QgsPostgresResult result( connectionRO()->PQexec( sql ) );
+  for ( int i = 0; i < result.PQntuples(); ++i )
+  {
+    if ( result.PQgetisnull( i, 2 ) ) continue; // config can be null and it's OK
+
+    const QString &configTxt = result.PQgetvalue( i, 2 );
+    const QString &type = result.PQgetvalue( i, 1 );
+    const QString &fname = result.PQgetvalue( i, 0 );
+    QVariantMap config;
+    QDomDocument doc;
+    if ( doc.setContent( configTxt ) )
     {
-      // CREATE TABLE qgis_editor_widget_styles (schema_name TEXT NOT NULL, table_name TEXT NOT NULL, field_name TEXT NOT NULL,
-      //                                         type TEXT NOT NULL, config TEXT,
-      //                                         PRIMARY KEY(schema_name, table_name, field_name));
-      QgsField &field = mAttributeFields[i];
-      const QString sql = QStringLiteral( "SELECT type, config FROM %1 WHERE schema_name = %2 and table_name = %3 and field_name = %4 LIMIT 1" ).
-                          arg( EDITOR_WIDGET_STYLES_TABLE, quotedValue( mSchemaName ), quotedValue( mTableName ), quotedValue( field.name() ) );
-      QgsPostgresResult result( connectionRO()->PQexec( sql ) );
-      for ( int i = 0; i < result.PQntuples(); ++i )
-      {
-        const QString type = result.PQgetvalue( i, 0 );
-        QVariantMap config;
-        if ( !result.PQgetisnull( i, 1 ) ) // Can be null and it's OK
-        {
-          const QString configTxt = result.PQgetvalue( i, 1 );
-          QDomDocument doc;
-          if ( doc.setContent( configTxt ) )
-          {
-            config = QgsXmlUtils::readVariant( doc.documentElement() ).toMap();
-          }
-          else
-          {
-            QgsMessageLog::logMessage( tr( "Cannot parse widget configuration for field %1.%2.%3\n" ).arg( mSchemaName, mTableName, field.name() ), tr( "PostGIS" ) );
-          }
-        }
+      config = QgsXmlUtils::readVariant( doc.documentElement() ).toMap();
+    }
+    else
+    {
+      QgsMessageLog::logMessage(
+        tr( "Cannot parse widget configuration for field %1.%2.%3\n" )
+        .arg( mSchemaName, mTableName, fname ), tr( "PostGIS" )
+      );
+      continue;
+    }
 
+    // Set corresponding editor widget
+    for ( auto &field : mAttributeFields )
```

Comment:
```suggestion
    for ( auto &field : qgis::as_const( mAttributeFields ) )
```

Reply:
field needs be non-const in order to be able to do field.setEditorWidgetSetup

Reply:
Most classes will detach and work on a copy of the container if iterating them non const.
Does this code here work as expected with QgsFields?

Reply:
As tests do exist and used to fail when I made a mistake I'd say it's working as-is